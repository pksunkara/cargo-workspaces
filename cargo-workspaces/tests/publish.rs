mod utils;
use insta::assert_snapshot;

#[test]
fn test_dry_run_single() {
    // `--allow-dirty` is supplied to not break tests during development.
    let mut err = utils::run_err(
        "../fixtures/single",
        &["ws", "publish", "--dry-run", "--allow-dirty"],
    );
    utils::normalize_output(&mut err);
    assert_snapshot!(err);
}

#[test]
fn test_dry_run_normal() {
    // `--allow-dirty` is supplied to not break tests during development.
    let mut err = utils::run_err(
        "../fixtures/normal",
        &["ws", "publish", "--dry-run", "--allow-dirty"],
    );
    utils::normalize_output(&mut err);
    assert_snapshot!(err);
}

#[test]
fn test_dry_run_normal_missing_fields() {
    // `--allow-dirty` is supplied to not break tests during development.
    let mut err = utils::run_err(
        "../fixtures/normal_missing_fields",
        &["ws", "publish", "--dry-run", "--allow-dirty"],
    );
    utils::normalize_output(&mut err);
    assert_snapshot!(err);
}
