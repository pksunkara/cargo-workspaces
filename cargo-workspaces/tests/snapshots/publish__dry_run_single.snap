---
source: tests/publish.rs
expression: err
---
warn Dry run doesn't check that all dependencies have been published. 
warn Dry run doesn't perform versioning. 
info checking simple
info success ok
